### PR TITLE
[BOO] Remove explicit prepadding from `use_custom=False` conv module

### PR DESCRIPTION
**Pull request by zjgarvey**
Changes were required to enable running the driver script with `torch.compile` as in #1134 , since the pre-pad was generating an additional memcopy. Since we don't use this module for any other purpose, it makes sense to revert it back to the fully default implementation.

### CODE REVIEW

File: iree/turbine/kernel/boo/op_exports/conv.py [hunk 1]
```diff
@@ -427,14 +427,10 @@ def __init__(self, sig: ConvSignature):
         self.kwargs = sig.get_conv_kwargs()
         if not sig.bias:
             self.kwargs["bias"] = None
-        self.explicit_padding = sig.explicit_padding
```
[the remark this answers is not in the document]
Comment (zjgarvey) [reply iree/turbine/kernel/boo/op_exports/conv.py:430]: I'd like to update other default `nn.Modules` for backward modes to eventually call `torch.ops.aten.convolution_backward` in a separate patch. The custom implementations will still need to use the explicit padding.

[the remark this answers is not in the document]
Comment (rkayaith) [reply iree/turbine/kernel/boo/op_exports/conv.py:430]: sounds good to me 👍
